lldb/docs/use/troubleshooting.rst

```diff
@@ -153,3 +153,148 @@ Linux applies to Windows. To find dependencies like Python, you need to run
 ``dumpbin`` on ``liblldb.dll`` too.
 
 On MacOS, use ``otool -l <path-to-lldb>``.
+
+Why Do I See More, Less, Or Different Registers Than I Expected?
+----------------------------------------------------------------
+
+The registers you see in LLDB are defined by information either provided by the
+debug server you are connected to, or in some cases, guessed by LLDB.
+
+If you are not seeing the registers you expect, the first step is to figure out
+which method is being used to read the register information. They are presented
+here in the order that LLDB will try them.
+
+Target Definition Script
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+These scripts tell LLDB what registers exist on the debug-server without having
+to ask it. You can check if you are using one by checking the setting:
+
+::
+
+   (lldb) settings show plugin.process.gdb-remote.target-definition-file
+
+In most cases you will not be using such a script.
+
+If you are using one, or want to write one, you can learn about them by reading
+the ``*_target_definition.py`` files in the
+`Python examples folder <https://github.com/llvm/llvm-project/tree/main/lldb/examples/python>`__.
+
+We recommend that before attempting to write a target definition script to solve
+your issues, you look into the other methods first.
```

Comment:
Would it make sense to make this an "[Admonition](https://docutils.sourceforge.io/docs/ref/rst/directives.html#admonitions)" ?

```suggestion
  .. tip:: 
  We recommend that before attempting to write a target definition script to solve
  your issues, you look into the other methods first.
```

Reply:
Done